src/API/Google/Settings.php

```diff
@@ -387,6 +387,58 @@ public function get_store_address(): AccountAddress {
 		return $mc_address;
 	}
 
+	/**
+	 * Check whether the address has errors
+	 *
+	 * @param AccountAddress $address to be validated.
+	 *
+	 * @return array
+	 */
+	public function has_address_errors( AccountAddress $address ): array {
+		if ( ! $address instanceof AccountAddress ) {
+			return [];
+		}
+
+		/** @var WC $wc */
+		$wc = $this->container->get( WC::class );
+
+		$country   = $address->getCountry();
+		$countries = $wc->get_wc_countries();
+
+		$locale          = $countries->get_country_locale();
+		$locale_settings = isset( $locale[ $country ] ) ? $locale[ $country ] : [];
+
+		$fields_to_validate = [
+			'address_1' => $address->getStreetAddress(),
+			'city'      => $address->getLocality(),
+			'country'   => $address->getCountry(),
+			'postcode'  => $address->getPostalCode(),
+			'state'     => $address->getRegion(),
+		];
+
+		return $this->validate_address( $fields_to_validate, $locale_settings );
+	}
+
+	/**
+	 * Check whether the required address fields are empty
+	 *
+	 * @param array $address_fields to be validated.
+	 * @param array $locale_settings locale settings
+	 * @return array
+	 */
+	public function validate_address( array $address_fields, array $locale_settings ) {
```

Comment:
💅  return type missed

Reply:
✅   [#7b1c89b](https://github.com/woocommerce/google-listings-and-ads/pull/1169/commits/7b1c89bf566dfd9c097f84ac6e9d256a54f7b584)